java/code/src/com/redhat/rhn/domain/action/ActionFactory.java

```diff
@@ -803,6 +803,7 @@ public static void rescheduleFailedServerActions(Action action, Long tries) {
         .setParameter("tries", tries)
         .setParameter("failed", ActionFactory.STATUS_FAILED)
         .setParameter("queued", ActionFactory.STATUS_QUEUED).executeUpdate();
+        updateActionEarliestDate(action);
```

Comment:
I think it does not make a difference practically, but you first re-schedule and after it you change the date.
Logically it would be better to do it the other way around. First change the date and than re-schedule the action. Otherwise a race condition may happen